From 912c0b5e962ccc595957cf9a1b9224ac9dce8e7c Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@e-labworks.com>
Date: Thu, 30 Jul 2026 13:26:25 -0300
Subject: [PATCH] toradex: adjust ahab_status exit code behavior on ELE based
 SoCs

Adjust exit code behavior to return an error if any ELE event is found,
so that the result of the command can be checked from a boot script.

SoCs using the EdgeLock Enclave (e.g. i.MX9x) implement the ahab_status
command in ele_ahab.c, which reported the events found but always
returned success. This is the ELE counterpart of the same change
already done for the SECO based implementation used by i.MX8.

Upstream-Status: Inappropriate [Toradex specific]

Signed-off-by: Sergio Prado <sergio.prado@e-labworks.com>
---
 arch/arm/mach-imx/ele_ahab.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/ele_ahab.c b/arch/arm/mach-imx/ele_ahab.c
index acefc62b5266..d8e532592c02 100644
--- a/arch/arm/mach-imx/ele_ahab.c
+++ b/arch/arm/mach-imx/ele_ahab.c
@@ -649,7 +649,7 @@ static int do_ahab_status(struct cmd_tbl *cmdtp, int flag, int argc, char *const
 	for (i = 0; i < cnt; i++)
 		display_event(events[i]);
 
-	return 0;
+	return CMD_RET_FAILURE;
 }
 
 static int do_sec_fuse_prog(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
-- 
2.34.1

